### ISSUE TITLE
[Relay] Warning SSL certificate is about to expire

### ISSUE THREAD

**Issue by AStove** [weechat/weechat#1508]
## Feature description
TL;DR: Please give certificate expiration date in relay handshake response.

For the relay ssl connection, many users use either self signed or Let's Encrypt certificates. These need to be renewed and more often than not, users forget to automate the renewal. For Let's Encrypt warnings are emailed. For self-signed certificates however there is no expiration warning. One day it will be expired, and weechat will keep working untill it's restarted, at which point it will refuse to start the relay. Leaving the user stranded and then has to connect via other means to figure out what has happened.

For webapps there is no way the app code can directly get information about the certificate of the underlying SSL connection.

Could the handshake response be used to send some details about the used certificate, most importantly, the expiration date. The app could then use this to warn the user. "Certificate about to expire in 5 days, please renew".

Arguments against this feature could be:
- Security issue?: This would not be a security issue because this information is publicly available anyway, just not for apps that run in a standard browser.
- Weechat is not the one that should be concerning itself with these lower level things.

Edit: Or perhaps have hdata for this? I did not think of this before but this might be better. It'd be tricky to find the exact hdata for the currently used relay connection though.

**Comment by flashcode** [comment]
I'm OK to provide such info to the client.

However, I'm not sure the new "handshake" command is the right place to give this info, because this command has nothing to do with how you are connected to relay (SSL or not).

Hdata gives access to WeeChat internal structure, so I'm not sure that's doable with this either (but… in some cases WeeChat builds hdata on the fly).

I can propose a new relay info, which is very easy to read from a client.

What do you think?

**Comment by flashcode** [comment]
By the way, if this is an info, you have to be authenticated to retrieve the info.
Not sure if it's a problem in your workflow.

**Comment by AStove** [comment]
You mean it can be retrieved with the info command? Where can I see a list of possible infos?
https://weechat.org/files/doc/stable/weechat_relay_protocol.en.html#command_info

**Comment by flashcode** [comment]
Yes, the infos are in API doc: https://weechat.org/files/doc/devel/weechat_plugin_api.en.html#_info_get

**Comment by flashcode** [comment]
For now it can not be as there's no such info, but I can create a new info for this.

**Comment by flashcode** [comment]
And there's a bug in relay doc, the command info takes a name and optional arguments, I'll fix that.

**Comment by AStove** [comment]
That would work for me. 😊

**Comment by flashcode** [comment]
By the way do you need something else than the expiration date?
I could make a flexible info with arguments, to retrieve different certificate properties.

**Comment by AStove** [comment]
Thumbprint, issuer, subject, valid from maybe, I don't need it but that's the most important thing I guess.
Something else: current connected relay for example "ssl.weechat". This would then allow us to automatically reload the relay if it's renewed already but not loaded into memory of weechat.

**Comment by lorenzhs** [comment]
Meh, renewal should really be automated. I think it's bad smell to include properties of the transport in the application protocol. You can use any other certificate expiration monitoring service to check your relay cert. I don't think WeeChat or any relay client should be involved in that.

**Comment by flashcode** [comment]
@lorenzhs: I agree with you, maybe we should not add such things in the relay protocol itself.
